source/language/gates.rst

```diff
@@ -283,42 +285,93 @@ A gate modifier is a keyword that applies to a gate. A modifier
 on the same or larger Hilbert space. We include modifiers in OpenQASM
 both for programming convenience and compiler analysis.
 
-The modifier ``inv @`` replaces its gate argument :math:`U` with its inverse
-:math:`U^\dagger`. The inverse of any gate can be defined recursively by
-reversing the order of the gates in its definition and replacing each of
-those with their inverse. The base case is given by replacing ``inv @ CX`` with ``CX`` and
-``inv @ U(θ, ϕ, λ)`` by ``U(-θ, -λ, -ϕ)``.
+The modifier ``ctrl @`` replaces its gate argument :math:`U` by a
+controlled-:math:`U` gate. If the control bit is 0, nothing happens to the target bit.
+If the control bit is 1, :math:`U` acts on the target bit. Mathematically, the controlled-:math:`U`
+gate is defined as :math:`C_U = I \otimes U^c`, where :math:`c` is the integer value of the control
+bit and :math:`C_U` is the controlled-:math:`U` gate. The new control qubit is prepended to the
+argument list for the controlled-:math:`U` gate. The modified gate does not use any additional
+scratch space and may require compilation to be executed. A special case is the controlled phase
+gate, which is defined as :math:`ctrl @ gphase(a) = U(0,0,a)`.
```

Comment:
I don't think controlled phase is a special case. In `stdgates.inc` we have:
```
gate cphase(lambda) a, b { ctrl @ phase(lambda) a, b; }
```
which properly resolves to a 2-qubit gate.

Reply:
This is copied exactly from pp. 14 of https://arxiv.org/pdf/2104.14722.pdf. In "The control modifier" paragraph. Happy to remove though

Reply:
That sentence in the paper appears to define a 1-qubit gate, not a 2-qubit `cphase`. @awcross1 can you confirm?

Reply:
Ah. Ok I think I interpreted it wrong. `ctrl @ gphase(lambda)` is just the 1q phase gate (although a bit odd bc control using implies two qubits).

Reply:
That's why I thought it was a "special case"

Reply:
That's right, it's a 1-qubit gate. The paper should say "controlled *global* phase gate" to avoid the confusion with the cphase gate.

Reply:
Addressed in https://github.com/Qiskit/openqasm/pull/203/commits/4471686f9e8f72a249b267a99622e1a86150777b.